cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/CouplingDeviceDiagramData.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.cgmes.extensions;
+
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.Switch;
+import com.powsybl.iidm.network.TwoWindingsTransformer;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * @author Massimo Ferraro <massimo.ferraro@techrain.eu>
+ */
+public class CouplingDeviceDiagramData<T extends Identifiable<T>> extends AbstractExtension<T> {
```

Comment:
Should we update this to use the new extension mechanism?

Reply:
yes